File: eFood.Model/Narudzba.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eFood.Model
{
    public class Narudzba
    {
        public int Id { get; set; }
        public DateTime DatumNarudzbe { get; set; }
        public int KorisnikId { get; set; }
        //public string Korisnik { get; set; }
        public int? DostavljacId{get; set; }
        public int StatusNarudzbeId { get; set; }
        public string StatusNarudzbe { get; set; }
        public string? StateMachine { get; set; }
        public string? PaymentId { get; set; }
    }
}

File: UI/efood_admin/lib/models/narudzba.dart
import 'package:json_annotation/json_annotation.dart';
part 'narudzba.g.dart';

@JsonSerializable()
class Narudzba {
  int? id;
  String? datumNarudzbe;
  int? korisnikId;
  int? statusNarudzbeId;
  int? dostavljacId;
  String? stateMachine;
  String? paymentId;

  Narudzba(
    this.id,
    this.datumNarudzbe,
    this.korisnikId,
    this.statusNarudzbeId,
    this.dostavljacId,
    this.stateMachine,
    this.paymentId,
  );

  factory Narudzba.fromJson(Map<String, dynamic> json) => _$NarudzbaFromJson(json);

  Map<String, dynamic> toJson() => _$NarudzbaToJson(this);
}
